src/helpers.jl

```diff
@@ -0,0 +1,7 @@
+# Vendored from Pluto.jl#src/notebook/path helpers.jl
+# from https://github.com/JuliaLang/julia/pull/36425
+function detectwsl()
+    Sys.islinux() &&
+    isfile("/proc/sys/kernel/osrelease") &&
+    occursin(r"Microsoft|WSL"i, read("/proc/sys/kernel/osrelease", String))
+end
```

Comment:
I'd be more comfortable with some error handling. Thoughts on this?
```suggestion
function detectwsl()
    Sys.islinux() || return false
    try
        return occursin(r"Microsoft|WSL"i, read("/proc/sys/kernel/osrelease", String))
    catch err
        @debug "Could not read /proc/sys/kernel/osrelease." exception=(err, catch_backtrace())
    end
    return false
end
```

Reply:
Sure. I also just found out that there is a new way to do this: https://github.com/JuliaLang/julia/pull/57069/files

Reply:
Yeah, that's neat.
```
@static if isdefined(Sys, :detectwsl)
    const detectwsl = Sys.detectwsl
else
    # attribution
    function detectwsl()
        # We use the same approach as canonical/snapd do to detect WSL
        islinux() && (
            isfile("/proc/sys/fs/binfmt_misc/WSLInterop")
            || isdir("/run/WSL")
        )
    end
end
```
then?

Reply:
seems great!

Reply:
I took the liberty up updating the PR with this.